code/__DEFINES/actions.dm

```diff
@@ -41,3 +41,10 @@ DEFINE_BITFIELD(check_flags, list(
 #define UPDATE_BUTTON_BACKGROUND (1<<2)
 #define UPDATE_BUTTON_OVERLAY (1<<3)
 #define UPDATE_BUTTON_STATUS (1<<4)
+
+/// Takes in a typepath of a `/datum/action` and adds it to `src`.
+/// Only useful if you want to add the action and never desire to reference it again ever.
+#define GRANT_ACTION(typepath) do {\
+	var/datum/action/_ability = new typepath(src);\
+	_ability.Grant(src);\
+} while (FALSE)
```

Comment:
in an ideal world this could also handle blackboard keys but i think we will get a very fucking messy define very quickly for all the stuff that doesn't need or care for blackboard keys or ai controllers